Add elanfp to plugins

MichaelCheng04 opened the pull request:
Type of pull request:

- [x] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

Support updating firmware for Elan fingerprint sensors.

hughsie commented:
@MichaelCheng04 I've rewritten the code a little, and so could you please check it still works and if required push any fixups to this branch please. I see what you mean about it being not-quite-CFU.

superm1 reviewed (dismissed):
Once you check it for hughsie's changes squash and LGTM

MichaelCheng04 commented:
> @MichaelCheng04 I've rewritten the code a little, and so could you please check it still works and if required push any fixups to this branch please. I see what you mean about it being not-quite-CFU.

@hughsie  I fixed two issues on afaf9f8010cb5163bb27525d927de7f1fe6ffbd0 . Thanks !!
